lib/rialto/etl/service_client/entity_resolver.rb

```diff
@@ -22,7 +22,13 @@ def initialize
 
         def resolve(type, params)
           resp = conn.get(type, params)
-          return resp.body if resp.success?
+          if resp.success?
```

Comment:
How would you feel about a structure like the following:

```ruby
return case resp.status
       when 200..299
         RDF::URI.new(resp.body)
       when 404
         nil
       end
raise Error, "Entity resolver returned #{resp.status} for #{type} type and #{params} params."
```
:question:

Reply:
worksforme